### DISCUSSION TITLE
Display error unique id

### DISCUSSION THREAD

**Discussion by krzysztof-cislo** [remix-run/remix#9786] [Q&A]
I would like to generate unique id in handleError where the error would be logged together with this id. Next I want to display the generated id in error boundary so user can send only the id which will be used to find correct log entry.

Is it possible?

**Comment by Erij-Maherzia-BEN-BRAHIM** [comment]
you can do it this way, in your action function throw an error with your ID
```ts
 if (yourCondition) {
    throw json({ errorCode: `GENERATED_ID_${id}` });
  }
```
then display it in your error boundary
```ts
import { isRouteErrorResponse, useRouteError } from "@remix-run/react";

export function GeneralErrorBoundary() {
  const error = useRouteError();

 if (
    isRouteErrorResponse(error) &&
    error.data?.errorCode?.startsWith("GENERATED_ID")
  ) {
    return (
        <p>GENERATED ID: {error.data.errorCode.split("GENERATED_ID_")[1]} </p>
    );
  }
}
```

**Reply by krzysztof-cislo** [reply]
This is not what I want to achieve. I want to create generic exception handler which handles all unexpected exceptions in single place.

Remix has two places to handle exceptions. 
* One is [`handleError`](https://remix.run/docs/en/main/file-conventions/entry.server#handleerror) on the server side. Here I would like to log exception all the details and produce just unique id
* Second one is [`ErrorBoundary`](https://remix.run/docs/en/main/route/error-boundary) where I can display this unique id to the user with information that something bad happened. I don't want to print all the details to the user.

This way, when user give me exception id I can find more information in the logs.
